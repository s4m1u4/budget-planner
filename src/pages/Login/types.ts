import { IUserAuthenticationData } from "../../types";

export interface LoginProps {
  onSubmit: (values: IUserAuthenticationData) => void;
  userAuthentication: (values: IUserAuthenticationData) => void | string;
}
