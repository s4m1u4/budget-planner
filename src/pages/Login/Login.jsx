import React from "react";
import { LoginForm } from "./components";
import { Box, Container } from "@mui/material";

import { box } from "./Login.styles";

export const Login = ({ userAuthentication }) => {
  return (
    <Container>
      <Box sx={box}>
        <LoginForm userAuthentication={userAuthentication} />
      </Box>
    </Container>
  );
};
