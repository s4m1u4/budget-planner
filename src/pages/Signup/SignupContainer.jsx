import { Signup } from "./Signup";
import { inject, observer } from "mobx-react";

export const SignupContainer = inject(
  ({
    rootStore: {
      userStore: { userRegistration },
    },
  }) => ({ userRegistration })
)(observer(Signup));
