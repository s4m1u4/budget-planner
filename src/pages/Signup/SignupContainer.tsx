import { Signup } from "./Signup";
import { compose } from "recompose";
import { inject, observer } from "mobx-react";
import { SignupProps } from "./types";

export const SignupContainer = compose<SignupProps, {}>(
  inject(
    ({
      rootStore: {
        userStore: { userRegistration },
      },
    }) => ({ userRegistration })
  ),
  observer
)(Signup);
