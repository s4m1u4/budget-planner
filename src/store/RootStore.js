import API from "../services/API";
import UserStore from "./UserStore";

export default class RootStore {
  API = new API();
  userStore = new UserStore();
}
