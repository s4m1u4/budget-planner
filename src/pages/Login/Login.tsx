import React, { FC } from "react";
import { LoginForm } from "./components";
import { Box, Container } from "@mui/material";
import { LoginProps } from "./types";

import { box } from "./Login.styles";

export const Login: FC<LoginProps> = ({ userAuthentication }) => {
  return (
    <Container>
      <Box sx={box}>
        <LoginForm userAuthentication={userAuthentication} />
      </Box>
    </Container>
  );
};
