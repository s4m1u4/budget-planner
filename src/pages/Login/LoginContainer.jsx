import { Login } from "./Login";
import { inject, observer } from "mobx-react";

export const LoginContainer = inject(
  ({
    rootStore: {
      userStore: { userAuthentication, isLoading },
    },
  }) => ({ userAuthentication, isLoading })
)(observer(Login));
