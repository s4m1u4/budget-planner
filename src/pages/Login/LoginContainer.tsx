import { Login } from "./Login";
import { compose } from "recompose";
import { inject, observer } from "mobx-react";
import { LoginProps } from "./types";

export const LoginContainer = compose<LoginProps, {}>(
  inject(
    ({
      rootStore: {
        userStore: { userAuthentication },
      },
    }) => ({ userAuthentication })
  ),
  observer
)(Login);
