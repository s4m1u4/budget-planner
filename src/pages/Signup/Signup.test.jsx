import { Provider } from "mobx-react";
import { RootStore } from "../../store/RootStore";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import { LoginContainer } from "../Login";
import { SignupContainer } from "./SignupContainer";
import userEvent from "@testing-library/user-event";

const getTitle = () =>
  screen.getByRole("heading", {
    name: /sign up/i,
  });
const getLinkToSignIn = () =>
  screen.getByRole("link", {
    name: /sign in/i,
  });
const getButtonSignIn = () =>
  screen.queryByRole("button", {
    name: /sign in/i,
  });

const renderWithStoreAndRouter = (initialEntries) => {
  const rootStore = new RootStore();

  return render(
    <Provider rootStore={rootStore}>
      <MemoryRouter initialEntries={initialEntries}>
        <Routes>
          <Route path="/login" element={<LoginContainer />} />
          <Route path="/signup" element={<SignupContainer />} />
        </Routes>
      </MemoryRouter>
    </Provider>
  );
};

describe("Sign up page", () => {
  beforeEach(() => {
    renderWithStoreAndRouter(["/signup"]);
  });

  it("should render the title", () => {
    expect(getTitle()).toBeInTheDocument();
  });

  it("should have link to 'Log in page'", () => {
    expect(getLinkToSignIn()).toBeInTheDocument();
  });

  it("should be changed to the 'Sign in page'", () => {
    expect(getButtonSignIn()).not.toBeInTheDocument();

    userEvent.click(getLinkToSignIn());

    expect(getButtonSignIn()).toBeInTheDocument();
  });
});
