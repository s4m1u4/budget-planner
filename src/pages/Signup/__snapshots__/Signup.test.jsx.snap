// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up page make snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-60biy9"
      >
        <div
          class="MuiBox-root css-o8u0d7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
          >
            Sign up
          </h1>
          <form
            class="MuiBox-root css-hbh1ay"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-1"
                id="mui-1-label"
              >
                First name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-1"
                  name="firstName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      First name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-2"
                id="mui-2-label"
              >
                Last name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  name="lastName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Last name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-3"
                id="mui-3-label"
              >
                Email
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-3"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-4"
                id="mui-4-label"
              >
                Password
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-4"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Sign up
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
          
          <div
            class="MuiBox-root css-10kxgn8"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Already have an account? 
              <a
                href="/login"
              >
                Sign in
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-60biy9"
      >
        <div
          class="MuiBox-root css-o8u0d7"
        >
          <h1
            class="MuiTypography-root MuiTypography-h6 css-1kgxqm0-MuiTypography-root"
          >
            Sign up
          </h1>
          <form
            class="MuiBox-root css-hbh1ay"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-5"
                id="mui-5-label"
              >
                First name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-5"
                  name="firstName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      First name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-6"
                id="mui-6-label"
              >
                Last name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-6"
                  name="lastName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Last name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-7"
                id="mui-7-label"
              >
                Email
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-7"
                  name="email"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-8"
                id="mui-8-label"
              >
                Password
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-8"
                  name="password"
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Password
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-ke5b6m-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Sign up
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
          
          <div
            class="MuiBox-root css-10kxgn8"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Already have an account? 
              <a
                href="/login"
              >
                Sign in
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
